src/hist_file.rs

```diff
@@ -56,27 +56,32 @@ fn find_unescaped(contents: &[char], target: char) -> Option<usize> {
 /// ```
 fn remove_quoted_strings(contents: String, delimiter: char) -> String {
 
-    fn _remove_quoted_strings(contents: Vec<char>, delimiter: char) -> Vec<char> {
+    fn _remove_quoted_strings<'a>(contents: &'a[char], delimiter: char, slices: &mut Vec<&'a [char]>) {
         match find_unescaped(&contents, delimiter) {
-            Some(fist_match) => {
-                let ret = &contents[0..fist_match];
-                match find_unescaped(&contents[fist_match+1..], delimiter) {
+            Some(first_match) => {
+                let ret = &contents[0..first_match];
+                match find_unescaped(&contents[first_match+1..], delimiter) {
                     Some(second_match) => {
-                        let rest = &contents[fist_match+second_match+2..];
-                        let mut concat: Vec<char> = vec![];
-                        concat.extend_from_slice(ret);
-                        concat.extend_from_slice(rest);
-                        _remove_quoted_strings(concat, delimiter)
+                        slices.push(ret);
+                        let rest = &contents[first_match+second_match+2..];
+                        _remove_quoted_strings(rest, delimiter, slices);
                     },
-                    None => contents
+                    None => {return;}
```

Comment:
This feels a bit ugly but I don't know what is the idiomatic way to leave a function in a match statement.

Reply:
You could use the `if-let` syntax instead.

```rust
if let Some(first_match) = find_unescaped(&contents, delimiter) {
	let ret = &contents[0..first_match];
	if let Some(second_match) = find_unescaped(&contents[first_match+1..], delimiter) {
		slices.push(ret);
                let rest = &contents[first_match+second_match+2..];
                _remove_quoted_strings(rest, delimiter, slices);	}
}

Reply:
Thank you for the tip. It is indeed way better.